File: backend/__SLUG__/src/domains/health/controller.go
package health

import (
	"encoding/json"
	"net/http"

	"go.uber.org/zap"
	"go_backend/src/logger"
)

// Controller handles HTTP interactions for health domain
type Controller struct {
	service *Service
	mapper  *Mapper
}

// NewController creates a new controller instance
func NewController(service *Service, mapper *Mapper) *Controller {
	return &Controller{
		service: service,
		mapper:  mapper,
	}
}

// GetHealth handles GET /health requests
func (c *Controller) GetHealth(w http.ResponseWriter, r *http.Request) {
	// Get health from service (business logic layer)
	health, err := c.service.GetHealth()
	if err != nil {
		logger.LoggerFromContext(r.Context()).Error("health check error", zap.Error(err))
		http.Error(w, "internal server error", http.StatusInternalServerError)
		return
	}

	dto := c.mapper.ToDTO(health)

	w.Header().Set("Content-Type", "application/json")
	if err := json.NewEncoder(w).Encode(dto); err != nil {
		logger.LoggerFromContext(r.Context()).Error("health encode error", zap.Error(err))
		http.Error(w, "internal server error", http.StatusInternalServerError)
		return
	}
}

File: backend/__SLUG___node/src/domains/health/controller.ts
import type { Request, Response } from "express";
import { loggerFromContext } from "../../logger/logger";
import type { Service } from "./service";
import { toDTO } from "./mapper";

export class Controller {
  constructor(private readonly service: Service) {}

  getHealth = (_req: Request, res: Response): void => {
    try {
      const health = this.service.getHealth();
      res.status(200).json(toDTO(health));
    } catch (err) {
      loggerFromContext().error({ err }, "health check error");
      res.status(500).send("internal server error");
    }
  };
}
